### PR TITLE
add tool to visualize inference model

### PR DESCRIPTION
**Pull request by fc500110**

### CODE REVIEW

File: paddle/fluid/inference/utils/visualization.cc [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+#include "paddle/fluid/platform/init.h"
+
+DEFINE_string(model_dir, "", "model path");
+DEFINE_string(model_program, "", "model program path");
+DEFINE_string(model_params, "", "model params path");
+
+USE_PASS(graph_viz_pass);
+USE_PASS(graph_to_program_pass);
+
+int main(int argc, char *argv[]) {
```
Comment (panyx0718) [inline paddle/fluid/inference/utils/visualization.cc]: Write some comments to say what's the usage of this tool?

File: paddle/fluid/inference/utils/visualization.cc [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      return -1;
+    } else {
+      argument.SetModelProgramPath(FLAGS_model_program);
+      argument.SetModelParamsPath(FLAGS_model_params);
+    }
+  } else {
+    argument.SetModelDir(FLAGS_model_dir);
+  }
+
+  argument.SetIrAnalysisPasses({"graph_viz_pass"});
```
Comment (panyx0718) [inline paddle/fluid/inference/utils/visualization.cc]: where is the graph file generated?

File: paddle/fluid/inference/utils/visualization.cc [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+#include <fstream>
+#include <memory>
+#include "paddle/fluid/framework/ir/graph_viz_pass.h"
+#include "paddle/fluid/inference/analysis/analyzer.h"
+#include "paddle/fluid/inference/analysis/argument.h"
+#include "paddle/fluid/inference/analysis/passes/ir_analysis_pass.h"
+#include "paddle/fluid/platform/init.h"
+
+DEFINE_string(model_dir, "", "model path");
+DEFINE_string(model_program, "", "model program path");
```
Comment (Superjomn) [inline paddle/fluid/inference/utils/visualization.cc]: model_program_path
model_params_path

File: paddle/fluid/inference/utils/visualization.cc [hunk 4]
```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
+//
```
Comment (Superjomn) [inline paddle/fluid/inference/utils/visualization.cc]: better to change filename to  visualizer

File: paddle/fluid/inference/utils/visualization.cc (hunk 1 above)
Comment (fc500110) [reply paddle/fluid/inference/utils/visualization.cc]: I have added usage.

File: paddle/fluid/inference/utils/visualization.cc (hunk 2 above)
Comment (fc500110) [reply paddle/fluid/inference/utils/visualization.cc]: Graph file is generated in current directory, and default graph filename is "0_ir_origin.dot". I have added comments in source code.

**Review by Superjomn — APPROVED** [review]
LGTM

**Pull request merged**